app/prices.js

```diff
@@ -449,3 +451,48 @@ function remove (sku, emit) {
 
     return match;
 }
+
+function organizePrices (prices) {
+    // Organize prices in an object, this way we will only have to loop through the items with matching attributes
+    const sorted = {};
+    for (let i = 0; i < prices.length; i++) {
+        if (prices[i].buy === null) {
+            continue;
+        }
+
+        const attribs = getAttributes(prices[i].sku);
+
+        if (!sorted[attribs['quality']]) {
+            // Define object, if not yet defined
+            sorted[attribs['quality']] = {};
+        }
+
+        if (Array.isArray(sorted[attribs['quality']][attribs['killstreak']])) {
+            sorted[attribs['quality']][attribs['killstreak']].push(important(prices[i]));
+        } else {
+            sorted[attribs['quality']][attribs['killstreak']] = [important(prices[i])];
+        }
+    }
+
+    return sorted;
+}
+
+function getAttributes (sku) {
+    const obj = SKU.fromString(sku);
+
+    // We only need quality and killstreak, they are the two biggest groups
+    return {
+        quality: obj.quality,
+        killstreak: obj.killstreak
+    };
+}
+
+function important (item) {
+    // Return only the data we need
+    return {
+        name: item.name,
+        buy: item.buy,
+        sell: item.sell,
+        time: item.time
+    };
+}
```

Comment:
I am not sure if this function is needed

Reply:
or would improve performance

Reply:
It should save us some amount of memory, since without it we would have to save over 20000*2 useless properties.

Reply:
I think it would use even more memory because now you have the original pricelist and the new one saved in memory.

Reply:
Without that function it would still have both of them, except the newly created one would use more memory.